### PR TITLE
HASS MQTT Discovery

### PR DESCRIPTION
**Pull request by yozik04**
Working on #82 

We need to think how to configure it.

### CODE REVIEW

**Comment by yozik04** [comment]
MQTT Discovery for inputs and outputs now works. At least switches work great in my HomeAssistant.

PS: My brain is struggling against functional programming. Doh... Especially `globals`

File: pi_mqtt_gpio/server.py [hunk 1]
```diff
@@ -641,7 +647,75 @@ def gpio_interrupt_callback(module, pin):
     )
 
 
+def hass_annonce_digital_input(in_conf, topic_prefix, mqtt_config):
```
Comment (flyte) [inline pi_mqtt_gpio/server.py]: typo in 'announce' here, as well as on line 683 (and the usages of the functions)

Comment (yozik04) [reply pi_mqtt_gpio/server.py]: fixed

**Comment by yozik04** [comment]
Basically it is working and is configurable now. Do you want me to change anything?

**Comment by yozik04** [comment]
So what we do with this pull request?

**Comment by flyte** [comment]
Sorry @yozik04 I missed the emails about your messages. I'll take a look at this over the weekend and get it merged in. Thanks!

**Comment by flyte** [comment]
Looks great. Thanks for implementing this!

**Pull request merged**